crates/biome_js_analyze/src/lint/nursery/use_spread.rs

```diff
@@ -0,0 +1,149 @@
+    /// The spread operator `...` can be used to achieve the same result, which is more concise and easier to read.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// foo.apply(null, args);
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// foo.apply(undefined, args);
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// obj.foo.apply(obj, args);
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// foo(...args);
+    ///
+    /// obj.foo(...args);
+    ///
+    /// foo.apply(obj, [1, 2, 3]);
+    ///
+    /// foo.apply(null, [1, 2, 3]);
+    /// ```
+    ///
+    pub UseSpread {
+        version: "next",
+        name: "useSpread",
+        language: "js",
+        sources: &[RuleSource::Eslint("prefer-spread").same()],
+        recommended: true,
+        fix_kind: FixKind::Safe,
+    }
+}
+
+impl Rule for UseSpread {
+    type Query = Ast<JsCallExpression>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = UseSpreadOptions;
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        let callee = node.callee().ok()?;
+
+        let member_expr = AnyJsMemberExpression::cast(callee.into_syntax())?;
+        if member_expr.member_name()?.text() != "apply" {
+            return None;
+        }
+
+        let arguments = node.arguments().ok()?.args();
+        if arguments.len() != 2 {
+            return None;
+        }
+
+        let first_arg = arguments.first()?.ok()?;
+        let this_arg = first_arg.as_any_js_expression()?;
+
+        let second_arg = arguments.last()?.ok()?;
+        let spread_candidate = second_arg.as_any_js_expression()?;
+
+        // The rule should not flag `.apply()` calls where the second argument is an array literal.
+        if spread_candidate.syntax().kind() == JsSyntaxKind::JS_ARRAY_EXPRESSION {
+            return None;
+        }
+
+        let applied_object = member_expr.object().ok()?;
+
+        let is_this_correct = if let Some(object_member) =
+            AnyJsMemberExpression::cast(applied_object.clone().into_syntax())
+        {
+            object_member.object().ok()?.syntax().text_trimmed() == this_arg.syntax().text_trimmed()
+        } else {
+            // This handles cases like `foo.apply(null, args)` or `foo.apply(undefined, args)`
+            this_arg
+                .as_static_value()
+                .is_some_and(|v| v.is_null_or_undefined())
+        };
+
+        if is_this_correct { Some(()) } else { None }
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, _state: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+        Some(RuleDiagnostic::new(
+            rule_category!(),
+            node.range(),
+            markup! {
+                "Use the spread operator instead of "<Emphasis>".apply()"</Emphasis>"."
+            },
```

Comment:
Please reword the diagnostics to match our rule pillars: https://biomejs.dev/linter/#rule-pillars

Also "use the..." Sentence is already part of the action, so it's redundant here